shift_detector/precalculations/EmbeddingDistancePrecalculation.py

```diff
@@ -0,0 +1,67 @@
+from shift_detector.precalculations.TextEmbeddingPrecalculation import TextEmbeddingPrecalculation
+from shift_detector.precalculations.Store import Store
+from shift_detector.precalculations.Precalculation import Precalculation
+from datawig.utils import random_split
+from math import sqrt
+
+
+class EmbeddingDistancePrecalculation(Precalculation):
+
+    def __init__(self, model=None, trained_model=None):
+        self.model = model
+        self.trained_model = trained_model
+
+    def __eq__(self, other):
+        return self.model == other.model \
+               and self.trained_model == other.trained_model
+
+    def __hash__(self):
+        return hash((self.model, self.trained_model))
+
+    @staticmethod
+    def calculate_distance(vector1, vector2):
+        delta_vector = [0] * len(vector1)
+        for idx, val in enumerate(vector1):
+            delta_vector[idx] = vector1[idx] - vector2[idx]
+        distance = 0
+        for i in delta_vector:
+            distance += i ** 2
+        return sqrt(distance)
+
+    @staticmethod
+    def join_and_normalize_vectors(series):
+        vector = [0.0] * len(series.iloc[0])
+        for cell in series:
+            for idx, val in enumerate(cell):
+                vector[idx] += float(val)
+        count = len(series)
+        for idx, val in enumerate(vector):
+            vector[idx] = val / count
+        return vector
+
+    def process(self, store: Store) -> dict:
+        """
+        Calculate the Sørensen dice coefficient between two columns
+        :param store:
+        :return: CheckResult
+        """
+
+        df1, df2 = store[TextEmbeddingPrecalculation(model=self.model, trained_model=self.trained_model)]
+
+        df1a, df1b = random_split(df1, [0.95, 0.05], seed=11)           # Baseline for df1
```

Comment:
Do not hard code the seed. If you need a seed for replicability in the test. Set them in the test ...